### PR TITLE
SRCH-101

### PR DESCRIPTION
**Pull request by scott-lewis-nsidc**
Editing bounding box text values will redraw the box.  Box will be retained when switching views.  Global box will display properly when it crosses the dateline.

### CODE REVIEW

**Comment by scott-lewis-nsidc** [comment]
As of this writing, the "audit" check is failing in CircleCI due to a vulnerability in the `async` package.  There looks to be some pull requests that will fix those packages (hopefully soon).  The vulnerability itself only affects DEV, as it affects the development web server stuff.

File: src/lib/SearchMap.js [hunk 1]
```diff
@@ -250,7 +251,19 @@ export default class SearchMap {
      *
      */
     globalCoordinatesFromExtent(west, south, east, north) {
-        return [[west, south], [east, south], [east, north], [west, north], [west, south]];
+        return [[[west, south], [east, south], [east, north], [west, north], [west, south]]];
```
Comment (juliacollins) [inline src/lib/SearchMap.js:254]: So this now needs an array of arrays of arrays? Wow!

Comment (juliacollins) [reply src/lib/SearchMap.js:254]: Ah, I think I see the connection to the changes in function calls now.

File: src/lib/SearchMap.js [hunk 2]
```diff
@@ -259,7 +272,7 @@ export default class SearchMap {
     polarCoordinatesFromExtent(projection, west, south, east, north) {
         let box = multisegmentBoxFromExtent(west, south, east, north);
         let globalToPolar = getTransform('EPSG:4326', projection);
-        return _.chunk(globalToPolar(box), 2);
+        return [_.chunk(globalToPolar(box), 2)];
```
Comment (juliacollins) [inline src/lib/SearchMap.js:275]: Is the array response from polarGeometryFromGlobalExtent still necessary, given that an array is now returned here? (See line 298.)

File: src/lib/SearchMap.js [hunk 3]
```diff
@@ -303,16 +316,22 @@ export default class SearchMap {
...
+
+    /**
+     *  Create the actual box "layer" for display on the map
+     */
+    createExtentLayer(geometry) {
         this.extentLayer = new VectorLayer({
             source: new VectorSource({ features: [
-                new Feature({geometry})
-            ] }),
+                    new Feature({geometry})
```
Comment (juliacollins) [inline src/lib/SearchMap.js:330]: Formatting seems off.

File: src/lib/SearchMap.js (hunk 1 above)
Comment (scott-lewis-nsidc) [reply src/lib/SearchMap.js:254]: Yeah, before the "outer" array was wrapped by the caller, not sure why.  Polygons have an array of array of coordinates - the first of the "outer" array is the polygon itself, and subsequent entries are "holes" in the polygon.  We aren't using holes so we just don't provide the other entries.

MultiPolygons have one more level, as each is a separate polygon.

File: src/lib/SearchMap.js (hunk 3 above)
Comment (scott-lewis-nsidc) [reply src/lib/SearchMap.js:330]: I'll see what happened and fix it.

File: src/lib/SearchMap.js (hunk 2 above)
Comment (scott-lewis-nsidc) [reply src/lib/SearchMap.js:275]: Let me take a look, I may be getting lost in all the nested array stuff...

**Comment by scott-lewis-nsidc** [comment]
Looks like the `async` module was patched!

**Review by juliacollins — APPROVED** [review]

**Pull request merged**